test-data/unit/fine-grained.test

```diff
@@ -426,6 +426,201 @@ main:3: error: Module has no attribute "g"
 ==
 main:3: error: Module has no attribute "g"
 
+[case testIgnoreWorksAfterUpdate]
+import a
+[file a.py]
+import b
+int() + str()  # type: ignore
+[file b.py]
+x = 1
+[file b.py.2]
+x = 2
+[file b.py.3]
+x = 3
+[delete b.py.4]
+[out]
+==
+==
+==
+a.py:1: error: Cannot find module named 'b'
+a.py:1: note: (Perhaps setting MYPYPATH or using the "--ignore-missing-imports" flag would help)
+
+[case testIgnoreWorksWithMissingImports]
+import a
+[file a.py]
+import b
+import xyz  # type: ignore
+xyz.whatever
+[file b.py]
+x = 1
+[file b.py.2]
+x = 2
+[file b.py.3]
+x = 3
+[file xyz.py.4]
+[out]
+==
+==
+==
+a.py:3: error: "object" has no attribute "whatever"
+
+[case testAddedIgnoreWithMissingImports]
+import a
+[file a.py]
+from b import x
+y: int = x
+[file b.py]
+from xyz import x
+[file b.py.2]
+from xyz import x  # type: ignore
+[file xyz.py.3]
+x = str()
+[out]
+b.py:1: error: Cannot find module named 'xyz'
+b.py:1: note: (Perhaps setting MYPYPATH or using the "--ignore-missing-imports" flag would help)
+==
+==
+a.py:2: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+
+[case testRemovedIgnoreWithMissingImport]
+import a
+[file a.py]
+from b import x
+y: int = x
+[file b.py]
+from xyz import x  # type: ignore
+[file b.py.2]
+from xyz import x
+[file xyz.py.3]
+x = str()
+[out]
+==
+b.py:1: error: Cannot find module named 'xyz'
+b.py:1: note: (Perhaps setting MYPYPATH or using the "--ignore-missing-imports" flag would help)
+==
+a.py:2: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+
+[case testRemovedModuleUnderIgnore]
+import a
+[file a.py]
+import c
+from b import x  # type: ignore
+y: int = x
+[file b.py]
+x = str()
+[file c.py]
+x = 1
+[delete b.py.2]
+[file c.py.3]
+x = 3
+[out]
+a.py:3: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+==
+==
+
+[case AddedModuleUnderIgnore]
+import a
+[file a.py]
+import c
+from b import x  # type: ignore
+y: int = x
+[file c.py]
+x = 1
+[file c.py.2]
+x = 2
```

Comment:
Again this doesn't trigger anything -- is it intentional?

Reply:
Yes, in both cases this is intentional. I decided to add few tests to check that `# type: ignore` works as expected after a no-op, just in case we will have a more complex logic for reprocessing errored targets in future.